library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -757,24 +911,55 @@ bool libspdm_asym_verify(
         if (!result) {
             return false;
         }
-        return libspdm_asym_verify_wrap(context, hash_nid, base_asym_algo,
-                                        param, param_size,
-                                        message_hash, hash_size,
-                                        signature, sig_size);
+        result = false;
+        if (try_big_endian) {
```

Comment:
It would be good to perform the try/fail only once. Once the endianness is determined then that can be stored in the context and would be used exclusively afterwards. Otherwise there is currently a performance hit for little endian, as it is always trying big endian first.

Reply:
@steven-bellock  This is an interesting optimization, In this case, should it try little endian then big endian? So the code would be something like...
```
result = false;
if (!context->try_little_endian_first && try_big_endian) {
    result = verify_in_big_endian(); 
     try_big_endian = false; 
}
if (!result && try_little_endian) {
    result = verify_in_little_endian(); 
    if (result) {
        context->try_little_endian_first = true
    }
}
if (!result && try_big_endian) {
    result = verify_in_big_endian(); 
    if (result) {
        context->try_little_endian_first = false;
    }
}
```

Reply:
Once the endianness is determined after one success verification, then we should stick to it, in this SPDM connection.

No need to detect again.